packages/arb-token-bridge-ui/src/util/EthDepositUtils.ts

```diff
@@ -0,0 +1,29 @@
+import { EthBridger } from '@arbitrum/sdk'
+import { Provider } from '@ethersproject/providers'
+import { BigNumber, constants } from 'ethers'
+import { DepositGasEstimates } from '../hooks/arbTokenBridge.types'
+
+export async function depositEthEstimateGas({
+  amount,
+  walletAddress,
+  l1Provider,
+  l2Provider
+}: {
+  amount: BigNumber
+  walletAddress: string
+  l1Provider: Provider
+  l2Provider: Provider
+}): Promise<DepositGasEstimates> {
+  const ethBridger = await EthBridger.fromProvider(l2Provider)
+
+  const depositRequest = await ethBridger.getDepositRequest({
+    amount,
```

Comment:
i think we don't need to pass the amount here as it's only for getting the gas estimation?

Reply:
Interesting. Added a ticket for this. Open for further discussion - https://github.com/OffchainLabs/arbitrum-token-bridge/issues/880

Reply:
We have to if we're gonna use `eth_estimateGas` in the background. Otherwise we'll have to hardcode values.

Reply:
We have to if we want to use the estimate gas methods from the SDK that use `eth_estimateGas` calls. Otherwise we'd have to hardcode the values.